src/query.ts

```diff
@@ -7,27 +11,36 @@ async function run(): Promise<void> {
   try {
     const query = getInput("query", { required: true });
     const language = getInput("language", { required: true });
-    const nwo = getInput("repository", { required: true });
+    const nwos: string[] = JSON.parse(
+      getInput("repositories", { required: true })
+    );
     const token = getInput("token", { required: true });
     const codeql = getInput("codeql", { required: true });
 
     setSecret(token);
 
-    // 1. Use the GitHub API to download the database using token
-    const dbZip = await downloadDatabase(token, nwo, language);
-    await unbundleDatabase(dbZip);
+    const curDir = cwd();
+    for (const nwo of nwos) {
+      const safeNwo = nwo.replace("/", "#");
+      const workDir = mkdtempSync(path.join(curDir, safeNwo));
```

Comment:
I initially tried using just `safeNwo`, but it was much easier to test this way because then I could query the same repository over and over. We could instead delete `workDir` after uploading the artifacts.

Reply:
Doesn't seem to be any harm in this - so long as it uses the correct directory when it needs to I don't think anyone would have reason to mind. Cleaning up wastes time that an ephemeral instance shouldn't need to spend until after it's work is done.

Reply:
Even a self-hosted runner should clear up anything in the working directory, so I think it's fine to create whatever files/directories you want in there.

As I understand it, `mkdtempSync(path.join(curDir, safeNwo))` is creating a directory in the tmp directory, but using `path.join(curDir, safeNwo)` as a prefix and then also appending 6 random characters. So this won't ever be reusing the same directory, but the directories it creates will have very verbose names.

Is this what was intended? Have I misunderstood?

Reply:
No, I think I'm wrong. It does make a directory in `curDir`, using `safeNwo` as a prefix for the final directory name. So that's fine and you can ignore that part of my comment above.

And I see now that by "it was much easier to test this way because then I could query the same repository over and over" you meant so it avoids reusing the same directory so you don't have to delete it. I didn't understand initially.

Reply:
We might discover in the future that deletion is better because we run out of disk space, but we can cross that bridge when we come to it.